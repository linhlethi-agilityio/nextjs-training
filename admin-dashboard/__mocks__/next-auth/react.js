export const useSession = () => ({
  data: {
    user: {
      token: 'access-token',
      email: 'user@gmail.com',
      id: '1',
      name: 'User',
      role: 'admin',
    },
    expires: Date.now().toString(),
  },
  status: 'authenticated',
  update: async () => null,
});
