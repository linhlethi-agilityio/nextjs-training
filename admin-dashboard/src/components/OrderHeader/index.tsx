'use client';

import { useCallback, useState } from 'react';
import { Flex, Text } from '@chakra-ui/react';
import { usePathname, useRouter } from 'next/navigation';

// Components
import { Dropdown } from '@/components';

// Constants
import { OPTIONS_PAGINATION } from '@/constants';

interface OrderHeaderProps {
  limit?: number;
  page?: number;
}

const OrderHeader = ({ limit = 10, page = 1 }: OrderHeaderProps) => {
  const [value, setValue] = useState<number>(limit);

  const pathname = usePathname();
  const { replace } = useRouter();

  const startResult = (page - 1) * limit + 1;
  const endResult = page * limit;

  /**
   * Function handle change limit page
   */
  const handleChangeLimitPage = useCallback(
    (value: string) => {
      setValue(Number(value));
      replace(`${pathname}?limit=${value}`);
    },
    [pathname, replace],
  );

  return (
    <Flex pl={7} mb={4} justifyContent="space-between" alignItems="center">
      <Text color="textDefault">
        Showing result {startResult}-{endResult} Result
      </Text>
      <Flex alignItems="center" gap={15}>
        <Text color="textDefault">Item per page</Text>
        <Dropdown
          value={value}
          color="textDark"
          options={OPTIONS_PAGINATION}
          onChangeValue={handleChangeLimitPage}
          h={26}
          w={70}
        />
      </Flex>
    </Flex>
  );
};

export default OrderHeader;
