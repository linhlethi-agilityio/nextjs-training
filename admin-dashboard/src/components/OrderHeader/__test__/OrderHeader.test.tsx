import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import OrderHeader from '..';

const mockReplace = jest.fn();
const mockPathname = 'localhost:3000';

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  usePathname: jest.fn(() => mockPathname),
  useRouter: jest.fn(() => ({ replace: mockReplace })),
}));

const mockProps = {
  limit: 10,
};

describe('OrderHeader component', () => {
  it('should match snapshot for OrderHeader', () => {
    const { container } = render(<OrderHeader {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
