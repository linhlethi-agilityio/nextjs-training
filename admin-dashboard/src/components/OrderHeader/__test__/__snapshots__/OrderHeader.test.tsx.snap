// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHeader component should match snapshot for OrderHeader 1`] = `
<div>
  <div
    class="css-rvaklg"
  >
    <p
      class="chakra-text css-d4sxp6"
    >
      Showing result 
      1
      -
      10
       Result
    </p>
    <div
      class="css-15v75y9"
    >
      <p
        class="chakra-text css-d4sxp6"
      >
        Item per page
      </p>
      <div
        class="chakra-select__wrapper css-1owtwcz"
      >
        <select
          aria-label="Select"
          class="chakra-select css-1gb6uvk"
        >
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
          <option
            value="50"
          >
            50
          </option>
        </select>
        <div
          class="chakra-select__icon-wrapper css-1u2hnvd"
        >
          <svg
            aria-hidden="true"
            class="chakra-select__icon"
            focusable="false"
            role="presentation"
            style="width: 1em; height: 1em; color: currentColor;"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
